cfgov/core/tests/test_services.py

```diff
@@ -0,0 +1,75 @@
+import mock
+import icalendar
+from django.test import TestCase
+
+from core.services import ICSView
+
+
+class ICSViewTest(TestCase):
+
+    def setUp(self):
+        pass
+
+    @mock.patch('requests.get')
+    def test_get_event_json(self, mock_request_get):
+        """
+        Test that we're constructing the event source URL from which to fetch
+        JSON correctly.
+        """
+        # We want two possible responses, first a good, 200 response, and
+        # then a 404 response (a response that doesn't provide JSON). We
+        # need to make sure we're handling the ValueError (JSONDecodeError).
+        mock_good_response = mock.MagicMock()
+        mock_good_response.status_code = 200
+        mock_good_response.json.return_value = {'ics': {'some': 'json'}}
+
+        mock_bad_response = mock.MagicMock()
+        mock_bad_response.status_code = 404
+        mock_bad_response.json.side_effect = ValueError()
+
+        mock_request_get.side_effect = [
+            mock_good_response,
+            mock_bad_response
+        ]
+        view = ICSView(
+            event_source='http://localhost:9200/events/<event_slug>/')
+
+        source_status = view.get_event_json('myevent')
+        self.assertEqual(source_status, 200)
+        mock_request_get.assert_called_with(
+            'http://localhost:9200/events/myevent/')
+
+        source_status = view.get_event_json('myevent')
+        self.assertEqual(source_status, 404)
+        self.assertEqual(view.event_json, {})
+
+    def test_generate_ics(self):
+        """
+        Test that, given a specific set of JSON, the generate_ics
+        function generates valid iCalendar data.
+        """
+        view = ICSView(
+            event_source='http://localhost:9200/events/<event_slug>/')
```

Comment:
Use settings in `.env` https://github.com/cfpb/cfgov-refresh/blob/flapjack/.env_SAMPLE#L22-L23